src/test/java/bio/terra/service/snapshotbuilder/SnapshotRequestDaoTest.java

```diff
@@ -0,0 +1,205 @@
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles({"google", "unittest"})
+@Tag(Unit.TAG)
+@EmbeddedDatabaseTest
+class SnapshotRequestDaoTest {
+
+  @Autowired private DaoOperations daoOperations;
+  @Autowired private ProfileDao profileDao;
+  @Autowired private GoogleResourceDao resourceDao;
+  @Autowired private SnapshotRequestDao snapshotRequestDao;
+
+  private BillingProfileModel billingProfile;
+  private GoogleProjectResource projectResource;
+  private Dataset dataset;
+  private SnapshotAccessRequest snapshotAccessRequest;
+
+  private static final String EMAIL = "user@gmail.com";
+
+  @BeforeEach
+  void setUp() throws IOException {
+    BillingProfileRequestModel profileRequest = ProfileFixtures.randomBillingProfileRequest();
+    billingProfile = profileDao.createBillingProfile(profileRequest, "testUser");
+
+    projectResource = ResourceFixtures.randomProjectResource(billingProfile);
+    UUID projectId = resourceDao.createProject(projectResource);
+    projectResource.id(projectId);
+
+    dataset = daoOperations.createMinimalDataset(billingProfile.getId(), projectId);
+
+    snapshotAccessRequest = SnapshotBuilderTestData.ACCESS_REQUEST;
+  }
+
+  private SnapshotAccessRequestResponse create() {
+    return snapshotRequestDao.create(dataset.getId(), snapshotAccessRequest, EMAIL);
+  }
+
+  private void verifyResponseContents(SnapshotAccessRequestResponse response1) {
+    Assertions.assertNotNull(
+        response1.getId(), "Snapshot Access Request Response should have an id");
+    Assertions.assertNotNull(
+        response1.getCreatedDate(),
+        "Snapshot Access Request Response should have a create date timestamp");
+    assertThat(
+        "Snapshot Access Request Response should contain the same datasetId as the example",
+        response1.getDatasetId(),
+        equalTo(dataset.getId()));
+    assertThat(
+        "Snapshot Access Request Response should contain the same name as the example",
+        response1.getSnapshotName(),
+        equalTo(SnapshotBuilderTestData.RESPONSE.getSnapshotName()));
+    assertThat(
+        "Snapshot Access Request Response should contain the same research purpose as the example",
+        response1.getSnapshotResearchPurpose(),
+        equalTo(SnapshotBuilderTestData.RESPONSE.getSnapshotResearchPurpose()));
+    assertThat(
+        "Snapshot Access Request Response should contain the same snapshot builder request as the example",
+        response1.getSnapshotSpecification(),
+        equalTo(SnapshotBuilderTestData.BUILDER_REQUEST));
+    assertThat(
+        "Snapshot Access Request Response should contain the same user email as the example",
+        response1.getCreatedBy(),
+        equalTo(EMAIL));
+    assertThat(
+        "New Snapshot Access Request Response should have submitted status",
+        response1.getStatus(),
+        equalTo(SnapshotAccessRequestResponse.StatusEnum.SUBMITTED));
+  }
+
+  @Test
+  void getByIdTest() {
```

Comment:
I think TDR has as mix of conventions for test names. Personally don't think it's necessary to add `Test` to the name of a test method. My preferred convention is to name the test using the name of the method being tested, which is what intellij suggests by default when creating a JUnit 5 test from a class.
```
  void getById() {
  ...
  void getByIdNotFound() {
  ...
```